### PR TITLE
Add test and warning for duplicate Kafka topics

### PR DESCRIPTION
**Pull request by frant-hartm**
Relates to #19948

The issue seems to be fixed by
https://github.com/hazelcast/hazelcast-jet/pull/2732. Before this commit
this test reliably fails. After it I couldn't reproduce it with
various different partition counts etc.

Adding the test to catch any future regressions and a log warning that
duplicate topics were specified. This usually means mis-configuration.

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible

### CODE REVIEW

**Comment by frant-hartm** [comment]
run-lab-run

File: extensions/kafka/src/test/java/com/hazelcast/jet/kafka/impl/StreamKafkaPTest.java [hunk 1]
```diff
@@ -316,6 +316,42 @@ public void when_snapshotSaved_then_offsetsRestored() throws Exception {
...
+        int messageCount = 1000;
+        for (int i = 0; i < messageCount; i++) {
+            kafkaTestSupport.produce(topic, i, Integer.toString(i));
+        }
+
+        IList<Object> list = instances[0].getList("sink");
+        try {
+            assertTrueEventually(() -> assertThat(list).hasSize(messageCount), 15);
+            sleepSeconds(1);
+            assertThat(list).hasSize(messageCount);
```
Comment (ufukyilmaz) [inline extensions/kafka/src/test/java/com/hazelcast/jet/kafka/impl/StreamKafkaPTest.java:349]: Here, we may use `assertTrueAllTheTime` method:  https://github.com/hazelcast/hazelcast/blob/f34dba56c3349cb3c825bdebd2bc50bf99d28118/hazelcast/src/test/java/com/hazelcast/test/HazelcastTestSupport.java#L1176-L1185

**Review by ufukyilmaz — APPROVED** [review]
LGTM. I just added minor comment.

**Review by olukas — APPROVED** [review]
